brave/brave-ios
# Fix #5107: ERC20 Approve tx updates for transaction confirmation

## Summary of Changes
- Update header in `TransactionConfirmationView` and showing current allowance instead of transaction total
- Added `EditPermissionsView` for assigning
- Unit tests for `TransactionConfirmationStore` for erc20 approve transaction
- Updated ERC 20 Approve creation to use Unlimited as default allowance to match desktop, now that we display proposed allowance and allow custom allowance to be set. [Relevant desktop code](https://github.com/brave/brave-core/blob/master/components/brave_wallet_ui/common/hooks/swap.ts#L455).

This pull request fixes #5107 

## Submitter Checklist:


## Test Plan:
To get a ERC 20 Approve transaction:
1. Open & unlock wallet
2. Switch to Ropsten network
3. If you have 0 DAI, swap some ETH to DAI
4. Open swap panel and try to swap DAI to ETH. Swap button should show 'Activate Token DAI' if you have never swapped DAI before. Tap it.
5. Confirm the Approve transaction
6. Open Accounts tab
7. Tap account to see account activity. Should see ETH, USDC, DAI now, and DAI approve tx should be shown as 'Approve x DAI'


## Screenshots:
![unlimited confirmation](https://user-images.githubusercontent.com/5314553/162535961-f369a4b1-294c-4c14-82f3-8e8140399e20.png)

![unlimited edit permissions](https://user-images.githubusercontent.com/5314553/162536070-bbe97570-a8ef-47e9-823c-fd8418791631.png)

![fixed confirmation](https://user-images.githubusercontent.com/5314553/162535976-5e966976-fed6-4311-834e-4616f37aa57b.png)
ebc2a.png)

![fixed edit permissions](https://user-images.githubusercontent.com/5314553/162535982-637e91cc-6e34-4015-b4e6-f650ee5a45c2.png)



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
